### DISCUSSION TITLE
The WebUI is amazing

### DISCUSSION THREAD

**Discussion by zaydogan** [balzack/databag#121] [General]
I must admit, I am slowly falling in love with the databag web ui.
It is so simple but functional. Everything works. I love it. 
Also the translation quality is very good. It is not perfect, but pretty good. 
I will help in that regard. I also noticed that the translation on iOS is different and of lower quality. I can help streamline both. I can also add a new language as a native speaker.
When typing emojis on the web ui (you can just left-click and select emojis on a macbook) the size of a single emoji should be double. 
I think all popular messengers are doing that (double size) and when you type more than 3 emojis, they are normal size.
I love it that you can make voice and video calls. I have my own turn server and it works great.
What I miss at the moment is: 
- on the web ui, a ringing sound when you call someone. It is silent at the moment. 
- incoming call integration on iOS. This is provided by iOS (VoIP CallKit) and databag voice calls can be treated like normal phone calls. You pick-up and talk.
- notification sound for notifications on iOS. It is silent at the moment. (I can provide a custom sound that I can record on my piano, for example. I would appreciate, if the notification sound would be non-standard, so you can differentiate it by sound / ear.
- badges on iOS (number of unread messages) if possible.

Otherwise great job, great project. Keep it up !

**Comment by balzack** [comment]
Thank you for the kind words!

The sound notifications and ringing sound should be easy additions. The other two bullet points might be more involved. I will need to do research on how CallKit works, and the badge number needs to come from the server which makes it less trivial based on the current design.

My main focus for the last couple months, and for the next few months is to create a typescript SDK for the underlying data communication. While doing this, also refactor the mobile and web app to use that npm SDK package. The should greatly simplify the effort in creating new apps for the network. There are several other apps I plan to build.

That being said, I am happy to pause that effort for bug fixes and high priority features. I'll look at the emoji size and the message font color this week. Let me know what you think the priority is of the bullet points your bring up. The sound notifications for example has been brought up by other users as well.

The feedback I get from people like you goes a long ways...thank you!

**Reply by zaydogan** [reply]
You're welcome.
I think the most important one is the emoji size, although i don't know how difficult it can be.
Second most important font color fix.
Third most important is the notification sounds and ringing sounds.
Least important: Badge numbers (probably also the most difficult).

**Reply by zaydogan** [reply]
and pinch and zoom for images for the ios app would be nice.

**Reply by balzack** [reply]
yes, the pinch & zoom is already in my backlog. it shouldn't be too difficult.

**Comment by zaydogan** [comment]
I think i found a solution for the big emoji, at least on the web ui.
There is a font size selector. if I select large, the emoji is slightly bigger.
Maybe you can utilize that to show single emojis bigger. Even bigger than large, extra large.

![image](https://github.com/user-attachments/assets/0aafdb69-6e59-48a4-a798-08b566b620bd)

You need to figure out if the input is a single emoji and then apply big font to it.
The emoji value range is huge and detecting them may be difficult. 
You can
`const emojiRegex = /\p{Emoji}/u;`
`emojiRegex.test('⭐'); // true`
but this may lead to false positives.
Here is the standard emoji list:
https://www.unicode.org/Public/emoji/1.0//emoji-data.txt
https://www.unicode.org/Public/emoji/2.0//emoji-data.txt

**Reply by balzack** [reply]
Thanks for the emoji list and the regex.
I am seeing some false positives in my testing, so I am not yet sure what the best approach is.

**Reply by zaydogan** [reply]
Note that [officially](https://unicode.org/Public/emoji/11.0/emoji-data.txt), 0123456789#* and other characters are emojis too, so the property escape you might want to use is not Emoji but rather Extended_Pictographic which denotes all the characters typically understood as emojis!

https://stackoverflow.com/questions/18862256/how-to-detect-emoji-using-javascript

**Reply by balzack** [reply]
Yes, I think that works.

I plan on using this regex on the message, and if it matches, double the font size. Let me know if you agree.
``` 
/^\p{Extended_Pictographic}$/u.test( );

**Reply by zaydogan** [reply]
Sounds good. Can you triple the size ?

**Reply by zaydogan** [reply]
Can you also add x.com (without www) formerly known as twitter.com to link identification.
Thank you.
